### DISCUSSION TITLE
Grafana Custom Query

### DISCUSSION THREAD

**Discussion by fellipedemoraes** [burningalchemist/sql_exporter#583] [Q&A] [answered]
Hi,

I opened a support ticket for Grafana because I need some custom metrics from a database.

They said it would be a bug, I'm not sure about that.

I configured an exporter.mssql in the alloy.config file and in a mssqlqueries.yml file I added some metrics (not the ones I need, just a test as stated in the documentation https://grafana.com/docs/alloy/latest/reference/components/prometheus/prometheus.exporter.mssql/#custom-metrics).

The Alloy service starts normally and does not show any errors but the metric does not appear.

Any tips?

Thanks!

![git](https://github.com/user-attachments/assets/6e82adc3-e47f-49c4-9f0e-1c122e7da38b)

**Comment by burningalchemist** [comment, marked as answer]
Hi @fellipedemoraes, 

It's not a bug for sure, it's the way Grafana Alloy handles the configuration. 

I was recently on a call with Grafana's customer. We found out that you also need to whitelist the custom metrics in the config. Usually it's somewhere in the rules where the regex and action is configured (just follow the pipeline). Otherwise the metric is just being filtered out (and hence there's no error nor data).

Once you added the custom metrics name to the regex, it should appear. Please have a look, and let me know if you still have an issue. I'm happy to assist, but it's an issue on the alloy/agent config. 👍

**Comment by fellipedemoraes** [comment]
Hi,

Thanks for your help, I'll post the solution here for anyone else who may have difficulty creating a custom query with alloy/prometheus.

![alloy-config](https://github.com/user-attachments/assets/eaae3b3f-53cf-4290-87d2-57554fe7f8c7)

![collector](https://github.com/user-attachments/assets/7a5f5b2e-382a-4e5c-945e-2faa551ad680)

![sucesso](https://github.com/user-attachments/assets/77cbdbc4-2a5e-46a6-84bb-b2b787c0d3dc)


BR
